### ISSUE TITLE
Compilation error  Microsoft.CodeAnalysis.CSharp.CSharpCompilation.Update

### ISSUE THREAD

**Issue by kant2002** [DevTeam/Pure.DI#7]
I did try 
1. create new console app 
2. copy code from ShroedingersCat sample.
3. Add reference to Pure.DI
and receive following error.
```
1>C:\Program Files\dotnet\sdk\7.0.100-preview.5.22267.11\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.RuntimeIdentifierInference.targets(216,5): message NETSDK1057: You are using a preview version of .NET. See: https://aka.ms/dotnet-support-policy
1>Skipping analyzers to speed up the build. You can execute 'Build' or 'Rebuild' command to run analyzers.
1>CSC : error DI002: System.ArgumentException: Inconsistent syntax tree features Parameter name: trees
    at Microsoft.CodeAnalysis.Compilation.SyntaxTreeCommonFeatures(IEnumerable`1 trees)
    at Microsoft.CodeAnalysis.CSharp.CSharpCompilation.Update(ReferenceManager referenceManager, Boolean reuseReferenceManager, SyntaxAndDeclarationManager syntaxAndDeclarations)
    at Pure.DI.Core.MetadataBuilder.Build(IExecutionContext executionContext)
    at Pure.DI.Core.SourceBuilder.Build(IExecutionContext executionContext)
    at Pure.DI.Core.Generator.Generate(IExecutionContext context)
1>D:\d\scratch\ConsoleApp1\ConsoleApp1\Program.cs(13,11,13,15): error CS0246: The type or namespace name 'Pure' could not be found (are you missing a using directive or an assembly reference?)
1>D:\d\scratch\ConsoleApp1\ConsoleApp1\Program.cs(14,18,14,22): error CS0246: The type or namespace name 'Pure' could not be found (are you missing a using directive or an assembly reference?)
```

after I rebuild project, everything start working.

**Comment by NikolayPianikov** [comment]
@kant2002 cpould you please share this solution?

**Comment by NikolayPianikov** [comment]
Also could you make sure this sample does not work for .NET 6 or even for latest .NET 7 preview 7.0.100-preview.4.22252.9

**Comment by kant2002** [comment]
I was in Visual Studio 17.3.0 Preview 1.1 if that matters.

**Comment by NikolayPianikov** [comment]
Thx, will try to reproduce

**Comment by NikolayPianikov** [comment]
Looks like sometime ParseOptions features contains some "Features" in dictionary and it should be consistent with Pure DI components ParseOptions

**Comment by NikolayPianikov** [comment]
@kant2002 could you please try [this version](https://www.nuget.org/packages/Pure.DI/1.1.41-beta1) after ~ 10 min, it is still publishing now

**Comment by NikolayPianikov** [comment]
> JAB0001

looks like this is an error from JAB library :)

**Comment by kant2002** [comment]
Yeah, but first one was without Jab, I was add him later on. So only first one error from this project. Would try to repro, but I see this error just one time, I cannot reproduce it. Definitely will try repro again.

**Comment by NikolayPianikov** [comment]
Ok, thx, theoretically I've fixed it in 1.1.41-beta1

**Comment by kant2002** [comment]
I again receive this error with latest version. Let me restart VS to make sure that nothing is cached.

**Comment by kant2002** [comment]
[ConsoleApp1.zip](https://github.com/DevTeam/Pure.DI/files/8892347/ConsoleApp1.zip)
Here the project which send your source generator hug of death.

**Comment by NikolayPianikov** [comment]
Ok, thx

**Comment by NikolayPianikov** [comment]
Sorry, cannot reproduce with  Visual Studio 17.3.0 Preview 1.1 and Pure.DI 1.1.40 and with the attached solution.
Could you please delete all directories `bin` and `obj` in your solution and build it from the command line like `dotnet build` from the root solution directory

**Comment by kant2002** [comment]
I cannot repro again. Even before that I remove bin/obj and was able to repro in VS. I recently read about VS aggressively cache source generator which sounds like a problem.

**Comment by kant2002** [comment]
I again see this error even in `1.1.41-beta1`, It happens this time after I add Microsoft.Extensions.DependencyInjection package manually via csproj, then building project, so I suspect that all of that related to adding Nuget packages somehow.

**Comment by kant2002** [comment]
`dotnet publish` works ok, but error still present in VS Error Info

**Comment by NikolayPianikov** [comment]
@kant2002 could you please try [1.1.41-beta2](https://www.nuget.org/packages/Pure.DI/1.1.41-beta2)

**Comment by NikolayPianikov** [comment]
I've published 1.1.41 with last fixes. I hope it will help
